### PR TITLE
python3Packages.fastparquet: 0.5.0 -> 0.6.3

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 0.6.3

`cramjam` is now required.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 129140` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python38Packages.cramjam</li>
    <li>python38Packages.datashader</li>
    <li>python38Packages.fastparquet</li>
    <li>python39Packages.cramjam</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129140` at 147ea738 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>python38Packages.cramjam</li>
    <li>python38Packages.datashader</li>
    <li>python38Packages.fastparquet</li>
    <li>python39Packages.cramjam</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/development/python-modules/cramjam/default.nix:37:15:
```
   |
37 |   patches = [ ./Cargo.lock.patch ];
   |               ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/fastparquet/default.nix [hunk 1]
```diff
@@ -1,27 +1,59 @@
...
+    cramjam
+    fsspec
+    lz4
+    numba
+    numpy
+    pandas
+    python-snappy
+    thrift
+    zstandard
+    zstd
```
Comment (risicle) [inline pkgs/development/python-modules/fastparquet/default.nix:43]: This is a lot of packages to propagate now. Upstream only lists five in its `requirements.txt`. Am I missing something?

**Comment by risicle** [comment]
`cramjam` on macos 10.15 fails with:

```
 = note: ld: library not found for -liconv
          clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
```

Providing it with `libiconv` in the `buildInputs` solves it.

(`fastparquet` itself though can't be tested because of broken `smbprotocol`)

**Comment by veprbl** [comment]
Also see #124862

**Comment by fabaff** [comment]
> Also see #124862

Yupp, this is a duplicate of #124862.

**Pull request closed without merging**